hmac.go

```diff
@@ -41,19 +41,26 @@ func (h hmacer) verifySignature(signature, payload string) (bool, error) {
 	return hmac.Equal([]byte(expectedSignature), []byte(signature)), nil
 }
 
-func (h hmacer) parseSignature(signatureKeyPair string) (string, error) {
-	if !strings.Contains(signatureKeyPair, "|") {
-		return "", SignatureError{"Signature-key pair does not contain |"}
+func (h hmacer) getMatchingSignature(signaturePairs string) (string, string) {
+	pairs := strings.Split(signaturePairs, "&")
+	for _, pair := range pairs {
+		split := strings.Split(pair, "|")
+		if len(split) == 2 && split[0] == h.publicKey {
+			return split[0], split[1]
```

Comment:
This function returns both the public key and the signature, but we don't really need the public key after matching, we just need the signature to do the comparison. How about we return the signature plus an error that might be set if we couldn't match?